### PR TITLE
Template keep array iterator

### LINKED ISSUE #6994: haxe.Template error with dce
Examples from the manual give out a runtime error due to `dce=std/full`
From this and all the below:
https://haxe.org/manual/std-template.html#sub-templates

https://try.haxe.org/#D7326

### PR DESCRIPTION
**Pull request by markknol**
Should close https://github.com/HaxeFoundation/haxe/issues/6994

### CODE REVIEW

**Comment by RealyUniqueName** [comment]
Would be great to have a test for this case :)

File: std/haxe/Template.hx [hunk 1]
```diff
@@ -65,6 +65,9 @@ class Template {
 	**/
 	public static var globals : Dynamic = {};
 
+	// To avoid issues with DCE, keep the array iterator.
+	@:keep var hxKeepArrayIterator = [].iterator();
```
Comment (RealyUniqueName) [inline std/haxe/Template.hx]: I think the proper way to keep array iterator for `haxe.Template` is to add `@:ifFeature(haxe.Template)` to the `Array.iterator` field.

Comment (markknol) [reply std/haxe/Template.hx]: In which file should that be added? 

Also (curious), why should the array iterator have knowledge about the Template class?

Comment (RealyUniqueName) [reply std/haxe/Template.hx]: I'm not sure if that should be added to the `std/Array.hx` or to each target Array implementation.
Maybe someone knows a better way. I just feel that the proposed behavior to create two useless objects per template instance is even worse.
Let's wait for the third opinion :)

Comment (markknol) [reply std/haxe/Template.hx]: I can make it `static` if that helps. But if `@:ifFeature` is the way to go, Ill (try to) change to that (when I know how) 👍

Comment (RealyUniqueName) [reply std/haxe/Template.hx]: You have a valid concern regarding Array-Template relations. I guess the "opposite" @:ifFeature meta should exist. Maybe @simn knows.

**Comment by Simn** [comment]
This is such a hack... 

It bothers me that this has `@:keep` because we shouldn't need that even with hacks. Could you check if this works if you replace the `@:keep` with `@:ifFeature("haxe.Template.run")`?

**Comment by markknol** [comment]
Yes that works, I tested here https://try.haxe.org/#b02bc

**Comment by RealyUniqueName** [comment]
Maybe we need a meta to describe a dependency for dce.
E.g.:
```haxe
@:dependencyFeature(Array.iterator)
public function run() {}
```

**Comment by Simn** [comment]
I was thinking about that too, but I'm not sure it's worth the complexity... For now I'm gonna merge this as a pragmatic fix.

**Pull request merged**